src/workerd/jsg/setup.h

```diff
@@ -42,21 +42,17 @@ kj::Own<v8::Platform> defaultPlatform(uint backgroundThreadCount);
 // library.
 class V8System {
  public:
-  // Use the default v8::Platform implementation, as if by:
+  // Uses the default v8::Platform implementation, as if by:
   //   auto v8Platform = jsg::defaultPlatform();
-  //   auto v8System = V8System(*v8Platform);
-  V8System();
-
-  // `flags` is a list of command-line flags to pass to V8, like "--expose-gc" or
+  //   auto v8System = V8System(*v8Platform, flags);
+  // (Optional) `flags` is a list of command-line flags to pass to V8, like "--expose-gc" or
   // "--single_threaded_gc". An exception will be thrown if any flags are not recognized.
-  explicit V8System(kj::ArrayPtr<const kj::StringPtr> flags);
-
-  // Use a possibly-custom v8::Platform implementation. Use this if you need to override any
-  // functionality provided by the v8::Platform API.
-  explicit V8System(v8::Platform& platform);
+  explicit V8System(kj::ArrayPtr<const kj::StringPtr> flags = nullptr);
 
   // Use a possibly-custom v8::Platform implementation, and apply flags.
-  explicit V8System(v8::Platform& platform, kj::ArrayPtr<const kj::StringPtr> flags);
+  explicit V8System(v8::Platform& platform,
+      kj::ArrayPtr<const kj::StringPtr> flags,
+      v8::Platform* defaultPlatformPtr);
```

Comment:
This needs some commentary explaining that `platform` is expected to be some sort of wrapper around a default platform, and you must pass in the default platform pointer so that the message loop can be pumped.

Although arguably it would be more reusable if instead the caller passed in a callback function to pump the message loop, so that the platform doesn't actually have to wrap the default platform at all.

Reply:
Passing a callback sounds better, since `DefaultPlatform` isn't exported by v8 as of now.